GCTC-NTGC/gc-digital-talent
# Bug - Fix theme colour mode logic

## 🗒️  Summary

There was some poor logic in the `ThemeProvider` which resulted in the app being stuck between modes. This fixes that issue.

Related: https://github.com/GCTC-NTGC/gc-digital-talent/pull/4245#discussion_r994652801

## 🧪 Testing

1. Run storybook `npm run storybook --workspace=common`
2. Open the "Theme Switcher" story
3. Change the theme using the switcher component and the storybook dropdown
4. Confirm that the theme does not get stuck when using the browser pref option
